DieteticConsultationAPI/Repositories/DietRepository.cs

```diff
@@ -37,24 +37,25 @@ public async Task<Diet> AddOrUpdate(Diet diet)
 
         public async Task Delete(int id)
         {
-            Diet diet = await _context
-                .Diets
-                .FirstOrDefaultAsync(d => d.Id == id);
-
-            _context.Diets.Remove(diet);
-            await _context.SaveChangesAsync();
+            if (await _context.Diets.FirstOrDefaultAsync(d => d.Id == id) is { } diet)
+            {
+                _context.Diets.Remove(diet);
+                await _context.SaveChangesAsync();
+            }
         }
```

Comment:
Add here additional info when diet does not exist for a particular id, you can throw an exception to inform the end user of such purpose

Reply:
added additional condition for deleted resources